test/integration/kueuectl/passthrough_test.go

```diff
@@ -0,0 +1,134 @@
+	})
+
+	ginkgo.DescribeTable("Pass-through commands",
+		func(oType string, namespaced bool, makeObject func(ns string) client.Object, getPath string, expectGet string, patch string, expectGetAfterPatch string) {
+			obj := makeObject(ns.Name)
+			gomega.Expect(k8sClient.Create(ctx, obj)).To(gomega.Succeed())
+			key := client.ObjectKeyFromObject(obj)
+
+			identityArgs := []string{oType, key.Name}
+			if namespaced {
+				identityArgs = append(identityArgs, "-n", key.Namespace)
+			}
+
+			// get it
+			args := append([]string{"get"}, identityArgs...)
+			args = append(args, fmt.Sprintf("-o=jsonpath='%s'", getPath))
+			cmd := exec.Command(kueuectlPath, args...)
+			setupEnv(cmd, kassetsPath, kubeconfigPath)
+			out, err := cmd.CombinedOutput()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%q", string(out))
+			gomega.Expect(string(out)).To(gomega.BeComparableTo(expectGet))
+
+			// patch it
+			if patch != "" {
+				args = append([]string{"patch"}, identityArgs...)
+				args = append(args, "--type=merge", "-p", patch)
+				cmd = exec.Command(kueuectlPath, args...)
+				setupEnv(cmd, kassetsPath, kubeconfigPath)
+				_, err = cmd.CombinedOutput()
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%q", string(out))
+
+				args = append([]string{"get"}, identityArgs...)
+				args = append(args, fmt.Sprintf("-o=jsonpath='%s'", getPath))
+				cmd = exec.Command(kueuectlPath, args...)
+				setupEnv(cmd, kassetsPath, kubeconfigPath)
+				out, err = cmd.CombinedOutput()
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%q", string(out))
+				gomega.Expect(string(out)).To(gomega.BeComparableTo(expectGetAfterPatch))
+			}
+			// delete it
+			args = []string{"delete", oType}
+			if namespaced {
+				args = append(args, "-n", key.Namespace)
+			}
+			args = append(args, key.Name)
+			cmd = exec.Command(kueuectlPath, args...)
+			setupEnv(cmd, kassetsPath, kubeconfigPath)
+			_, err = cmd.CombinedOutput()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%q", string(out))
+
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, key, obj)
+			}, util.Timeout, util.Interval).Should(testing.BeNotFoundError())
+		},
+		ginkgo.Entry("Workload", "workload", true, makePassThroughWorkload, "{.spec.active}", "'true'", `{"spec":{"active":false}}`, "'false'"),
+		ginkgo.Entry("Workload(short)", "wl", true, makePassThroughWorkload, "{.spec.active}", "'true'", `{"spec":{"active":false}}`, "'false'"),
+		ginkgo.Entry("Cluster Queue", "clusterqueue", false, makePassThroughCQ, "{.spec.stopPolicy}", "'None'", `{"spec":{"stopPolicy":"Hold"}}`, "'Hold'"),
+		ginkgo.Entry("Cluster Queue(short)", "cq", false, makePassThroughCQ, "{.spec.stopPolicy}", "'None'", `{"spec":{"stopPolicy":"Hold"}}`, "'Hold'"),
+		ginkgo.Entry("Local Queue", "localqueue", true, makePassThroughLQ, "{.spec.clusterQueue}", "'queue'", ``, ""),
+		ginkgo.Entry("Local Queue (short)", "lq", true, makePassThroughLQ, "{.spec.clusterQueue}", "'queue'", ``, ""),
```

Comment:
I'm happy with testing just one object, not to bloat the tests too much.

Reply:
Done 

/unhold